### PR TITLE
try to work sqllogic-test out

### PR DESCRIPTION
**Pull request by hanfei1991**
Right now:
- statement tests (4 failed)
- complete tests (2331771/5938120 tests failed)

still need to fix them, but at least it's runable locally. This pr will check how ci runs

### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
run sqllogic test

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 6a5b1a4830dba2fedbacc776639b3baaf580922f with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56078/6a5b1a4830dba2fedbacc776639b3baaf580922f/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6667641011">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56078/6a5b1a4830dba2fedbacc776639b3baaf580922f/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by qoega** [comment]
This is great for statements tests!
Last time when we checked with Sema we saw many errors due to our test harness more strict than sqllogictest default one:
- We fail when we get string and test requires integer. 
- We can get a query timeout and not handle it now
- There is no error type in result definition

Example of test that is incorrectly defined
```
diff --git a/test/evidence/slt_lang_aggfunc.test b/test/evidence/slt_lang_aggfunc.test
index 6531aca..eee7941 100644
--- a/test/evidence/slt_lang_aggfunc.test
+++ b/test/evidence/slt_lang_aggfunc.test
@@ -112,12 +112,12 @@ SELECT total(y) FROM t1
 ----
 0

-query I nosort
+query T nosort
 SELECT min(y) FROM t1
 ----
 0

-query I nosort
+query T nosort
 SELECT max(y) FROM t1
 ----
```

File: src/Parsers/ExpressionListParsers.cpp [hunk 1]
```diff
@@ -2604,6 +2604,12 @@ Action ParserExpressionImpl::tryParseOperand(Layers & layers, IParser::Pos & pos
         }
     }
 
+    /// ignore all heading plus
```
Comment (alexey-milovidov) [inline src/Parsers/ExpressionListParsers.cpp:2607]: heading -> leading

**Comment by alexey-milovidov** [comment]
Now, we need support for non-constant branches of `caseWithExpression`.
It can be implemented by rewriting to `multiIf` instead of `transform` similarly to `caseWithoutExpression`.

**Comment by alexey-milovidov** [comment]
We can merge these PRs incrementally, to see incremental improvements.

**Comment by alexey-milovidov** [comment]
```
milovidov-desktop :) SELECT CASE 5 WHEN number THEN number + 1 END FROM numbers(10)

SELECT caseWithExpression(5, number, number + 1, NULL)
FROM numbers(10)

Query id: 273a84d1-e28d-4352-aa0d-0e128802145d


Elapsed: 0.026 sec. 

Received exception:
Code: 44. DB::Exception: Argument at index 1 for function transform must be constant. (ILLEGAL_COLUMN)

milovidov-desktop :) SELECT CASE 5 WHEN 5 = number THEN number + 1 END FROM numbers(10)

SELECT caseWithExpression(5, 5 = number, number + 1, NULL)
FROM numbers(10)

Query id: 2587f6f7-7984-479b-9db4-2926ca84883e


Elapsed: 0.001 sec. 

Received exception:
Code: 44. DB::Exception: Argument at index 1 for function transform must be constant. (ILLEGAL_COLUMN)

milovidov-desktop :) SELECT CASE WHEN 5 = number THEN number + 1 END FROM numbers(10)

SELECT multiIf(5 = number, number + 1, NULL)
FROM numbers(10)

Query id: ce1e0573-1167-4d04-8721-a1c65bc83dca

┌─if(equals(5, number), plus(number, 1), NULL)─┐
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
│                                            6 │
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
│                                         ᴺᵁᴸᴸ │
└──────────────────────────────────────────────┘
```

**Pull request merged**